src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -207,6 +215,17 @@ private function findClasses(string $namespace, string $pattern, array $excludeP
             if ($r->isInstantiable() || $r->isInterface()) {
                 $classes[$class] = null;
             }
+
+            if (80000 > \PHP_VERSION_ID || !$autoconfigure) {
+                continue;
```

Comment:
(@ro0NL using threads to allow scoped discussions)

> is there any reason for the "auto discovered services only" actually?

Yes: the attribute must be parsed before compiler passes, and when classes are loaded. The DI container of Symfony doesn't load class definitions if it doesn't have to. When autodiscovery is enabled, we know we need to load the corresponding classes, to that's the proper hook.

Reply:
given

```
_defaults { autoconfigure: true }

Foo\: { prototype: 'path/to/foos' }
Bar\Baz: ~
```

conceptually i expect the attribute to take effect when it's defined on either `Bar\Baz` or `Foo\*`.

when it's defined on abstracts/interfaces i expect it to behave the same as `_instanceof` directives would.

Reply:
alternatively, what seems to be the route you're taking now: `#[ServicePrototype(...)]`

or `#[ServicePrototypeAutoconfig(...)]` to go fully explicit in terms of exposing this is enabled thru `autoconfigure: true` for prototyped services.